storage-rdma/src/main/java/com/ibm/crail/storage/rdma/client/RdmaStorageLocalEndpoint.java

```diff
@@ -51,37 +51,23 @@
 
 public class RdmaStorageLocalEndpoint implements StorageEndpoint {
 	private static final Logger LOG = CrailUtils.getLogger();
-	private String indexDirPath;
-	private ConcurrentHashMap<Integer, CrailBuffer> bufferMap;
-	private ConcurrentHashMap<Integer, RdmaBlockIndex> indexMap;
+	private ConcurrentHashMap<Long, CrailBuffer> bufferMap;
 	private Unsafe unsafe;
 	private InetSocketAddress address;
 	
 	public RdmaStorageLocalEndpoint(InetSocketAddress datanodeAddr) throws Exception {
 		LOG.info("new local endpoint for address " + datanodeAddr);
 		this.address = datanodeAddr;
-		this.bufferMap = new ConcurrentHashMap<Integer, CrailBuffer>();
-		this.indexMap = new ConcurrentHashMap<Integer, RdmaBlockIndex>();
+		this.bufferMap = new ConcurrentHashMap<Long, CrailBuffer>();
 		this.unsafe = getUnsafe();
-		this.indexDirPath = RdmaStorageServer.getIndexDirectory(datanodeAddr);
-		File indexDir = new File(indexDirPath);
-		ByteBuffer fileBuffer = ByteBuffer.allocate(CrailConstants.BUFFER_SIZE);
-		if (indexDir.exists()){
-			for (File indexFile : indexDir.listFiles()) {
-				FileInputStream indexStream = new FileInputStream(indexFile);
-				FileChannel indexChannel = indexStream.getChannel();
-				fileBuffer.clear();
-				indexChannel.read(fileBuffer);
-				fileBuffer.flip();
-				RdmaBlockIndex blockIndex = new RdmaBlockIndex();
-				blockIndex.update(fileBuffer);
-				File dataFile = new File(blockIndex.getPath());
+		String dataPath = RdmaStorageServer.getDatanodeDirectory(datanodeAddr);
+		File dataDir = new File(dataPath);
+		if (dataDir.exists()){
```

Comment:
What happens if the dataDir does not exist? No mappings i.e. read/write will fail? Shouldn't we throw an exception?